src/stdlib/parse_groks.rs

```diff
@@ -137,39 +144,88 @@ impl Function for ParseGroks {
                         keyword: "patterns",
                         expr,
                     })?
                     .try_bytes_utf8_lossy()
-                    .expect("grok pattern not bytes")
+                    .map_err(|_| function::Error::InvalidArgument {
+                        keyword: "patterns",
+                        value: format!("{expr_:?}").into(),
+                        error: "grok pattern should be a string",
+                    })?
                     .into_owned();
                 Ok(pattern)
             })
             .collect::<std::result::Result<Vec<String>, function::Error>>()?;
 
-        let aliases = arguments
+        let mut aliases = arguments
             .optional_object("aliases")?
             .unwrap_or_default()
             .into_iter()
             .map(|(key, expr)| {
+                let expr_ = expr.clone();
                 let alias = expr
                     .resolve_constant(state)
                     .ok_or(function::Error::ExpectedStaticExpression {
                         keyword: "aliases",
                         expr,
-                    })
-                    .map(|e| {
-                        e.try_bytes_utf8_lossy()
-                            .expect("should be a string")
-                            .into_owned()
-                    })?;
+                    })?
+                    .try_bytes_utf8_lossy()
+                    .map_err(|_| function::Error::InvalidArgument {
+                        keyword: "aliases",
+                        value: format!("{expr_:?}").into(),
+                        error: "alias pattern should be a string",
+                    })?
+                    .into_owned();
                 Ok((key, alias))
             })
             .collect::<std::result::Result<BTreeMap<KeyString, String>, function::Error>>()?;
 
+        let alias_sources = arguments
+            .optional_array("alias_sources")?
+            .unwrap_or_default()
+            .into_iter()
+            .map(|expr| {
+                let expr_ = expr.clone();
+                let path = expr
+                    .resolve_constant(state)
+                    .ok_or(function::Error::ExpectedStaticExpression {
+                        keyword: "alias_sources",
+                        expr,
+                    })?
+                    .try_bytes_utf8_lossy()
+                    .map_err(|_| function::Error::InvalidArgument {
+                        keyword: "alias_sources",
+                        value: format!("{expr_:?}").into(),
+                        error: "alias source should be a string",
+                    })?
+                    .into_owned();
+                Ok(path)
+            })
+            .collect::<std::result::Result<Vec<String>, function::Error>>()?;
+
+        for src in alias_sources {
+            let path = Path::new(&src);
+            let file = File::open(path).map_err(|_| function::Error::InvalidArgument {
```

Comment:
I wonder if the contents of these files can be leaked. Probably not.
@fuchsnj is there any debug function that can print `src_aliases`?

Reply:
I think this is fine.